ethclient/gethclient/gethclient_test.go

```diff
@@ -518,3 +560,303 @@ func testCallContractWithBlockOverrides(t *testing.T, client *rpc.Client) {
 		t.Fatalf("unexpected result: %x", res)
 	}
 }
+
+func testTraceTransaction(t *testing.T, client *rpc.Client) {
+	ec := New(client)
+
+	var testSuite = []struct {
+		txHash    common.Hash
+		config    *tracers.TraceConfig
+		expectErr error
+		expect    interface{}
+	}{
+		{
+			txHash:    testTransactionHash,
+			config:    nil,
+			expectErr: nil,
+			expect: map[string]interface{}{
+				"gas":         21000,
+				"failed":      false,
+				"returnValue": "",
+				"structLogs":  nil,
+			},
+		},
+		{
+			txHash: testTransactionHash,
+			config: &tracers.TraceConfig{
+				Config: &logger.Config{
+					EnableMemory: true,
+				},
+			},
+			expectErr: nil,
+			expect: map[string]interface{}{
+				"gas":         21000,
+				"failed":      false,
+				"returnValue": "",
+				"structLogs":  nil,
+			},
+		},
+		{
+			txHash: testTransactionHash,
+			config: &tracers.TraceConfig{
+				TracerConfig: json.RawMessage(`{"tracer":"callTracer"}`),
```

Comment:
No this is not right. `&tracers.TraceConfig{Tracer: "callTracer"}` is correct.
`TracerConfig` field is the configuration object for the given tracer.